backend/src/main/java/com/google/sps/notifs/EmailNotifier.java

```diff
@@ -161,16 +158,21 @@ public void notifyUser() throws MessagingException, GeneralSecurityException, IO
                 + getRecipientName()
                 + " Please Join your Ad-Lib meeting via the link below : \n"
                 + " http://meet.google.com/new");
-    NetHttpTransport HTTP_TRANSPORT = GoogleNetHttpTransport.newTrustedTransport();
-    Gmail service =
-        new Gmail.Builder(HTTP_TRANSPORT, JSON_FACTORY, getCredentials(HTTP_TRANSPORT))
-            .setApplicationName(APPLICATION_NAME)
-            .build();
     Message messageWithEmail = createMessageWithEmail(email);
-    service.users().messages().send("me", messageWithEmail).execute();
+    getService().users().messages().send("me", messageWithEmail).execute();
   }
 
-  private String getToEmail() {
+  public String getToEmail() {
     return toEmail;
   }
+
+  /** Getter function that returns the string representing the email recipients name . */
+  public String getRecipientName() {
+    return recipientName;
+  }
+
+  /** Getter function that returns the Gmail Service that emailNotifier depends on . */
+  public Gmail getService() {
+    return service;
+  }
```

Comment:
Will any of these methods be used by production code? I agree with the intention here to make the class easier to test but we need to balance that against making too much of the class' internals visible for testing. The ideal is for the class to be fully testable through its public API that the non-test code uses.